platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -430,7 +442,7 @@ public DataFileReader<Bucket<K>> endWriting() throws IOException {
                     IntObjectPair<BucketMutation<K>> keyValue = iterator.next();
                     final int bucketIndex = keyValue.getOne();
                     final BucketMutation<K> bucketMap = keyValue.getTwo();
-                    flushExecutor.execute(() -> readUpdateQueueBucket(bucketIndex, bucketMap, queue));
+                    getFlushExecutor().execute(() -> readUpdateQueueBucket(bucketIndex, bucketMap, queue));
```

Comment:
`getFlushExecutor()` call could be made into a `final` local variable outside the `while` loops. It could be just a `static synchronized` method then,

Reply:
Done